code/common/pcapdecoder.q

```diff
@@ -0,0 +1,87 @@
+\d .pcap
+
+// size of headers in bytes and dict of protocol code conversions
+globheader: 24;
+packetheader: 16;
+allcodes:(enlist 6)!(enlist `TCP);
+
+
+buildtable:{[file]
+ // initial x and byte cut points removed from array list to make table
+ // gettablerow is iterated over each datapacket, extracting data
+
+ data:1_ last each {[n]
+  filebytesize: count n;
+  gettablerow[n;]\[{y>(first x[0])+40}[;filebytesize];(),0]    
+  } read1 file;
+ 
+ data: update sym:` from data;
+ `time`sym xcols data
+ }
+
+
+gettablerow:{[n;x]  // data for a single row
+ time:     first gettime[n;x];
+ flags:    getflags[n;x];
+ protocol: getprotocol[n;x];
+ 
+ totallength: (enlist 2;enlist "h")1: datafromfile[n;x;18;2];
+ IPheader:    4*"J"$last string n[x[0]+globheader+packetheader+16];
+ TCPheader:   4* first "0123456789abcdef"?/:string n[x[0]+globheader+packetheader+48];
+ len:         (first first totallength - IPheader + TCPheader) mod 65536;
+
+ length: first raze ((enlist "h";enlist 2)1: n[x[0]+36 37]) mod 65536;
+ data:   datafromfile[n;x;length - len;len];
+
+ ips: getips[n;x]; 
+ src:  ips[0];
+ dest: ips[1];
+
+ info: getinfo[n;x];
+ srcport:  first info[0] mod 65536;
+ destport: first info[1] mod 65536;
+ seq:      first info[2] mod 4294967296;
+ ack:      first info[3] mod 4294967296;
+ win:      first info[4] mod 65536;
+ tsval:    first info[5] mod 4294967296;
+ tsecr:    first info[6] mod 4294967296;
```

Comment:
It might be neater to do all this `mod` stuff inside the `getinfo` function & return a dictionary from that function, then just join the extra keys below in line 50. Probably also worth having a comment explaining the "magic numbers" being used here.